### PR TITLE
Add `patch` and `cctools` dependency

### PR DESCRIPTION
**Pull request by kenodegard**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

Xref: conda/conda-build#4495

In addressing the flakiness of conda-build's patch tests we found that the default `patch` shipped on MacOS (2.5.8) is insufficient (need >=2.6).

### CODE REVIEW

**Comment by isuruf** [comment]
Thanks. I recently noticed that we need `cctools` for macos for `conda_build.utils.macho`. What do you think about adding that here?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,7 +24,8 @@ build:
 
 requirements:
   build:
-    - m2-patch           # [win]
+    - patch    >=2.6     # [not win]
+    - m2-patch >=2.6     # [win]
```
[the remark this answers is not in the document]
Comment (isuruf) [reply recipe/meta.yaml:28]: If this recipe itself had patches and was built with a conda-build without `patch` as a dependency, then the system `patch` would be picked up.

[the remark this answers is not in the document]
Comment (jakirkham) [reply recipe/meta.yaml:28]: That makes sense, but we are not applying any patches currently right?

[the remark this answers is not in the document]
Comment (isuruf) [reply recipe/meta.yaml:28]: Yes, not currently, but it's okay to have it here when we do in the future.

[the remark this answers is not in the document]
Comment (isuruf) [reply recipe/meta.yaml:28]: Ping on this

[the remark this answers is not in the document]
Comment (jakirkham) [reply recipe/meta.yaml:28]: SGTM

**Pull request merged**